--- a/components/crash/core/app/BUILD.gn
+++ b/components/crash/core/app/BUILD.gn
@@ -242,6 +242,7 @@ if (is_mac || is_android || is_linux ||
   # We build a chromium-specific crashpad_handler executable so that we can
   # define custom UserStreamDataSources.
   executable("chrome_crashpad_handler") {
+    output_name = "helium_crashpad_handler"
     sources = [ "chrome_crashpad_handler.cc" ]
 
     deps = [
--- a/components/crash/core/app/crashpad_linux.cc
+++ b/components/crash/core/app/crashpad_linux.cc
@@ -169,7 +169,7 @@ bool PlatformCrashpadInitialization(
     if (!base::PathService::Get(base::DIR_EXE, &handler_path)) {
       return false;
     }
-    handler_path = handler_path.Append("chrome_crashpad_handler");
+    handler_path = handler_path.Append("helium_crashpad_handler");
 
     // When --use-cros-crash-reporter is set (below), the handler passes dumps
     // to ChromeOS's /sbin/crash_reporter which in turn passes the dump to
--- a/chrome/browser/memory_details_linux.cc
+++ b/chrome/browser/memory_details_linux.cc
@@ -119,7 +119,7 @@ void MemoryDetails::CollectProcessData(
   ProcessData current_browser =
       GetProcessDataMemoryInformation(GetAllChildren(process_map, getpid()));
   current_browser.name = base::ASCIIToUTF16(version_info::GetProductName());
-  current_browser.process_name = u"chrome";
+  current_browser.process_name = u"helium";
 
   for (auto i = current_browser.processes.begin();
        i != current_browser.processes.end(); ++i) {
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -151,7 +151,7 @@ if (!is_android && !is_mac) {
   if (is_win) {
     _chrome_output_name = "initialexe/chrome"
   } else {
-    _chrome_output_name = "chrome"
+    _chrome_output_name = "helium"
   }
 
   executable("chrome_initial") {
@@ -1688,19 +1688,19 @@ if (is_linux || is_chromeos) {
       }
     }
     extract_symbols("chrome_symbols") {
-      binary = "$root_out_dir/chrome"
+      binary = "$root_out_dir/helium"
 
       if (current_cpu == "x86") {
         # GYP used "ia32" so keep that naming for back-compat.
-        symbol_file = "$root_out_dir/chrome.breakpad.ia32"
+        symbol_file = "$root_out_dir/helium.breakpad.ia32"
       } else {
-        symbol_file = "$root_out_dir/chrome.breakpad.$current_cpu"
+        symbol_file = "$root_out_dir/helium.breakpad.$current_cpu"
       }
 
       deps = [ ":chrome" ]
     }
     extract_symbols("chrome_crashpad_symbols") {
-      binary = "$root_out_dir/chrome_crashpad_handler"
+      binary = "$root_out_dir/helium_crashpad_handler"
 
       if (current_cpu == "x86") {
         # GYP used "ia32" so keep that naming for back-compat.
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -32,10 +32,8 @@ declare_args() {
 assert(is_linux)
 
 packaging_files_executables = [
-  "$root_out_dir/chrome",
-  "$root_out_dir/chrome_crashpad_handler",
-  "$root_out_dir/chrome_management_service",
-  "$root_out_dir/chrome_sandbox",
+  "$root_out_dir/helium",
+  "$root_out_dir/helium_crashpad_handler",
 ]
 
 packaging_files_shlibs = []
@@ -170,12 +168,12 @@ action("merge_rpm_dependencies") {
 }
 
 strip_binary("strip_chrome_binary") {
-  binary_input = "$root_out_dir/chrome"
+  binary_input = "$root_out_dir/helium"
   deps = [ "//chrome" ]
 }
 
 strip_binary("strip_chrome_crashpad_handler") {
-  binary_input = "$root_out_dir/chrome_crashpad_handler"
+  binary_input = "$root_out_dir/helium_crashpad_handler"
   deps = [ "//components/crash/core/app:chrome_crashpad_handler" ]
 }
 
